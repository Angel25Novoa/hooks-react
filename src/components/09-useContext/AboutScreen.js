import React, {useContext} from 'react'
import { UserContext } from './UserContext'

const AboutScreen = () => {
  const {user, setUser} = useContext(UserContext) 
  return (
    <div>
      <h1>About Screen</h1>
      <hr />
      <button className="btn btn-info"
      onClick={() => setUser({})}>
        Logout
      </button>
      <pre>{JSON.stringify(user)}</pre>
    </div>
  )
}

export default AboutScreen