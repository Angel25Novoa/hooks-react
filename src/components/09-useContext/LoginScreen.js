import React, {useContext} from 'react'
import { UserContext } from './UserContext'

const LoginScreen = () => {
  //obtener context
  const {user, setUser} = useContext(UserContext) 
  //obtener setUser
  return (
    <div>
      <h1>Login Screen</h1>
      <hr />
      <button className="btn btn-info"
      onClick={() => setUser({
        name: 'Juan',
        email:'',
        id: 2
      })}>
        Login
      </button>
      <pre>{JSON.stringify(user)}</pre>
    </div>
  )
}

export default LoginScreen