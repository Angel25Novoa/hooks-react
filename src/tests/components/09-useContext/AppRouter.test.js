import React from 'react'
import {mount} from 'enzyme'
import AppRouter from '../../../components/09-useContext/AppRouter'
import { UserContext } from '../../../components/09-useContext/UserContext'

describe('Pruebas en el componente AppRouter', () => {
  const user = {
    name: 'Jorge',
    email: '',
    id: 2
  }
  const wrapper = mount(
    <UserContext.Provider value={{user}}>
      <AppRouter />
    </UserContext.Provider>
    )
  test('debe de mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })
})