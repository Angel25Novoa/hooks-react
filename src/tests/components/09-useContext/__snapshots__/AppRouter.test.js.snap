// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AppRouter debe de mostrar el componente correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            exact={true}
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            exact={true}
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="nav-link"
            exact={true}
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre>
        {
   "name": "Jorge",
   "email": "",
   "id": 2
}
      </pre>
    </div>
  </div>,
]
`;
